pymchelper/readers/topas.py

```diff
@@ -0,0 +1,172 @@
+                                        'size': float(match.group(2)),
+                                        'unit': match.group(3)}
+            else:
+                return False
+        return True
+
+    def get_scorer_name(self, output_data):  # skipcq: PYL-R0201
+        """Get scorer name from the output file"""
+        name = ""
+        pattern = r"# Results for scorer: (\w+)"
+        match = re.search(pattern, output_data)
+        if match:
+            name = match.group(1)
+        return name
+
+    def get_scorer_unit_results(self, output_data):  # skipcq: PYL-R0201
+        """Get scoring quantity, unit and the scoring values (sum/mean/etc.) from the output file"""
+        scorers = ['DoseToMedium', 'DoseToWater', 'DoseToMaterial', 'TrackLengthEstimator',
+                   'AmbientDoseEquivalent', 'EnergyDeposit', 'Fluence', 'EnergyFluence',
+                   'StepCount', 'OpticalPhotonCount', 'OriginCount', 'Charge', 'EffectiveCharge',
+                   'ProtonLET', 'SurfaceCurrent', 'SurfaceTrackCount', 'PhaseSpace']
+        for scorer in scorers:
+            if scorer in output_data:
+                unit = ""
+                pattern = f"# {scorer}\\s(\\( .*? \\))?\\s*: (.*)"
+                match = re.search(pattern, output_data)
+                if match:
+                    unit = ""
+                    if match.group(1):
+                        unit = match.group(1)[2:-2]
+                    results = match.group(2).split()
+                    return scorer, unit, results
+        return "", "", []
+
+    def get_differential_axis(self, output_data):  # skipcq: PYL-R0201
+        """Check if the output file contains differential axis and get it from file if it does"""
+        if "# Binned by" in output_data:
+            pattern = r"# Binned by (.+?) in (\d+) bin[s ] of (\d+) (\w+) from ([\d.]+) (\w+) to ([\d.]+) (\w+)"
+            match = re.search(pattern, output_data)
+            if match:
+                binned_by = match.group(1)
+                num_bins = int(match.group(2))
+                unit = match.group(4)
+                min_val = float(match.group(5))
+                max_val = float(match.group(7))
+
+                return MeshAxis(n=num_bins, min_val=min_val, max_val=max_val,
+                                name=binned_by, unit=unit, binning=MeshAxis.BinningType.linear)
+        return None
+
+    def read_data(self, estimator):
+        """
+        Read the data from the file and store them in the provided estimator object.
+        Topas reader assumes that the input file is in the same directory as the output file
+        to extract the number of histories from it - otherwise the number of histories is set to 0.
+        """
+        # find the input file to extract the number of histories from it
+        num_histories = 0
+        for input_file in os.listdir(self.directory):
+            if input_file.endswith(".txt"):
```

Comment:
You can extract input file name ("parameter file") from the comment section of the output file. Also - try to check what happens if the output file is located in a different location that the input file.

Reply:
if input file is not in the same directory as output file, number of histories will be set to 0